File: 1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int count = 0;
        while (count < students.size())
        {
            if (students[0] != sandwiches[0])
            {
                int tmp = students[0];
                students.erase(students.begin());
                students.push_back(tmp);
                count++;
            }
            else
            {
                count = 0;
                students.erase(students.begin());
                sandwiches.erase(sandwiches.begin());
            }
        }
        return students.size();
    }
};

File: 1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int[] cnt = new int[2];
        for (int i : students) cnt[i]++;

        for (int i : sandwiches) {
            if (cnt[i] > 0) cnt[i]--;
            else {
                return cnt[0] > 0 ? cnt[0] : cnt[1];
            }
        }

        return 0;
    }
}